daprdocs/content/en/js-sdk-docs/js-client/_index.md

```diff
@@ -77,6 +77,22 @@ dapr run --app-id example-sdk --app-protocol grpc -- npm run start
 npm run start:dapr-grpc
 ```
 
+### Environment Variables
+
+You can use the `DAPR_HTTP_ENDPOINT` and `DAPR_GRPC_ENDPOINT` environment variables to set the Dapr Sidecar's HTTP and gRPC endpoints respectively. When these variables are set, the `daprHost` and `daprPort` don't have to be passed to the constructor, the client will parse them automatically out of the provided endpoints.
+
+```typescript
+import { DaprClient, CommunicationProtocol } from "@dapr/dapr";
+
+// Using HTTP, when DAPR_HTTP_ENDPOINT is set
+const client = new DaprClient({ daprHost, daprPort });
+
+// Using gRPC, when DAPR_GRPC_ENDPOINT is set
+const client = new DaprClient({ communicationProtocol: CommunicationProtocol.GRPC });
```

Comment:
Are we showing here that this is the equivalent of setting the environment variable?

Reply:
I wonder if `DAPR_GRPC_ENDPOINT` is set, should we also by-default use communication protocol GRPC. Also while instantiating the `DaprClient`, should we clearly log if we used env var to figure out host/port/protocol.

Reply:
Good point! I can add a log.
Regarding the automatic setting of the protocol, I think we can't do it, because, as per the [proposal](https://github.com/dapr/proposals/blob/main/0008-S-sidecar-endpoint-tls.md#design) we can have both `DAPR_GRPC_ENDPOINT` and `DAPR_HTTP_ENDPOINT` set at the same time.
> DAPR_GRPC_ENDPOINT and DAPR_HTTP_ENDPOINT can be set at the same time since some SDKs (Java, as of now) supports both protocols at the same time and app can pick which one to use.

Reply:
Right (and maybe this can be added to the proposal to unify SDK implementations), so if 
1. `DAPR_GRPC_ENDPOINT` is set, set protocol to gRPC
1. `DAPR_HTTP_ENDPOINT` is set, set protocol to HTTP 
1. Both are set, use HTTP
1. Finally, if function param is specified, that overrides everything
1. No function param or env var also means HTTP